src/execution/physical_plan/plan_create_index.cpp

```diff
@@ -43,78 +34,11 @@ unique_ptr<PhysicalOperator> PhysicalPlanGenerator::CreatePlan(LogicalCreateInde
-	}
-	new_column_types.emplace_back(LogicalType::ROW_TYPE);
-	select_list.push_back(make_uniq<BoundReferenceExpression>(LogicalType::ROW_TYPE, op.info->scan_types.size() - 1));
-
-	auto projection = make_uniq<PhysicalProjection>(new_column_types, std::move(select_list), op.estimated_cardinality);
-	projection->children.push_back(std::move(table_scan));
-
-	// filter operator for IS_NOT_NULL on each key column
-
-	vector<LogicalType> filter_types;
-	vector<unique_ptr<Expression>> filter_select_list;
-
-	for (idx_t i = 0; i < new_column_types.size() - 1; i++) {
-		filter_types.push_back(new_column_types[i]);
-		auto is_not_null_expr =
-		    make_uniq<BoundOperatorExpression>(ExpressionType::OPERATOR_IS_NOT_NULL, LogicalType::BOOLEAN);
-		auto bound_ref = make_uniq<BoundReferenceExpression>(new_column_types[i], i);
-		is_not_null_expr->children.push_back(std::move(bound_ref));
-		filter_select_list.push_back(std::move(is_not_null_expr));
-	}
-
-	auto null_filter =
-	    make_uniq<PhysicalFilter>(std::move(filter_types), std::move(filter_select_list), op.estimated_cardinality);
-	null_filter->types.emplace_back(LogicalType::ROW_TYPE);
-	null_filter->children.push_back(std::move(projection));
-
-	// determine if we sort the data prior to index creation
-	// we don't sort, if either VARCHAR or compound key
-	auto perform_sorting = true;
-	if (op.unbound_expressions.size() > 1) {
-		perform_sorting = false;
-	} else if (op.unbound_expressions[0]->return_type.InternalType() == PhysicalType::VARCHAR) {
-		perform_sorting = false;
-	}
-
-	// actual physical create index operator
-
-	auto physical_create_index =
-	    make_uniq<PhysicalCreateARTIndex>(op, op.table, op.info->column_ids, std::move(op.info),
-	                                      std::move(op.unbound_expressions), op.estimated_cardinality, perform_sorting);
-
-	if (perform_sorting) {
-
-		// optional order operator
-		vector<BoundOrderByNode> orders;
-		vector<idx_t> projections;
-		for (idx_t i = 0; i < new_column_types.size() - 1; i++) {
-			auto col_expr = make_uniq_base<Expression, BoundReferenceExpression>(new_column_types[i], i);
-			orders.emplace_back(OrderType::ASCENDING, OrderByNullType::NULLS_FIRST, std::move(col_expr));
-			projections.emplace_back(i);
-		}
-		projections.emplace_back(new_column_types.size() - 1);
-
-		auto physical_order = make_uniq<PhysicalOrder>(new_column_types, std::move(orders), std::move(projections),
-		                                               op.estimated_cardinality);
-		physical_order->children.push_back(std::move(null_filter));
-
-		physical_create_index->children.push_back(std::move(physical_order));
-	} else {
-
-		// no ordering
-		physical_create_index->children.push_back(std::move(null_filter));
-	}
+	D_ASSERT(op.children.size() == 1);
+	auto table_scan = CreatePlan(*op.children[0]);
 
-	return std::move(physical_create_index);
+	PlanIndexInput input(context, op, table_scan);
+	return index_type->create_plan(input);
```

Comment:
Can we throw an exception if `create_plan` is not set?

Reply:
I guess an assertion will do?

Reply:
I think an exception is better - calling a `nullptr` leads to a crash otherwise.